esmvalcore/preprocessor/_mask.py

```diff
@@ -73,37 +51,32 @@ def _get_fx_mask(fx_data, fx_option, mask_type):
 
 def _apply_fx_mask(fx_mask, var_data):
     """Apply the fx data extracted mask on the actual processed data."""
-    # Broadcast mask
-    var_mask = np.zeros_like(var_data, bool)
-    var_mask = np.broadcast_to(fx_mask, var_mask.shape).copy()
-
     # Apply mask across
     if np.ma.is_masked(var_data):
-        var_mask |= var_data.mask
+        fx_mask |= var_data.mask
```

Comment:
you 100% sure they have the same dims? Is there a check that checks the mask array dims and barks at the user if they're not the same? Plan for the worst, always :grin:

Reply:
Well the fx_var has been broadcasted to the var's shape when added, so I assumed they would be the same at this point. Is there a need for a further check if that's the case?

Reply:
no, I missed the broadcasting to shape step :)